cobbler/actions/buildiso/__init__.py

```diff
@@ -54,34 +81,83 @@ def __init__(self, api):
         self.distmap = {}
         self.distctr = 0
         self.logger = logging.getLogger()
-
-        self.profiles: List[str] = []
+        self.templar = templar.Templar(api)
         self.isolinuxdir = ""
 
+        # based on https://uefi.org/sites/default/files/resources/UEFI%20Spec%202.8B%20May%202020.pdf
+        # FIXME: PowerPC is missing from the table
+        self.efi_fallback_renames = {
+            "grubaa64": "bootaa64.efi",
+            "grubx64.efi": "bootx64.efi",
+        }
+
         # grab the header from buildiso.header file
         self.iso_template = (
             pathlib.Path(self.api.settings().iso_template_dir)
             .joinpath("buildiso.template")
             .read_text(encoding="UTF-8")
         )
+        self.isolinux_menuentry_template = (
+            pathlib.Path(api.settings().iso_template_dir)
+            .joinpath("isolinux_menuentry.template")
+            .read_text(encoding="UTF-8")
+        )
+        self.grub_menuentry_template = (
+            pathlib.Path(api.settings().iso_template_dir)
+            .joinpath("grub_menuentry.template")
+            .read_text(encoding="UTF-8")
+        )
+
+    def _find_distro_source(self, known_file: str, distro_mirror: str) -> str:
+        """
+        Find a distro source tree based on a known file.
 
-    def copy_boot_files(self, distro, destdir: str, new_filename: str = ""):
+        :param known_file: Path to a file that's known to be part of the distribution,
+            commonly the path to the kernel.
```

Comment:
Nitpick: We normally indent until we are on par with the tag we started the docstring from.

```suggestion
        :param known_file: Path to a file that's known to be part of the distribution,
                           commonly the path to the kernel.
```